integrationTests/longTests/txsFromGatheredRewards_test.go

```diff
@@ -0,0 +1,209 @@
+	t.Skip("this is a long test")
+
+	numOfNodes := 2
+	advertiser := integrationTests.CreateMessengerWithKadDht(context.Background(), "")
+	_ = advertiser.Bootstrap()
+	advertiserAddr := integrationTests.GetConnectableAddress(advertiser)
+
+	_ = logger.SetLogLevel("*:DEBUG,process:TRACE")
+
+	nodesMap := integrationTests.CreateProcessorNodesWithNodesCoordinator(numOfNodes, 1, 1, numOfNodes, 1, advertiserAddr)
+	setRewardParametersToNodes(nodesMap)
+
+	defer func() {
+		_ = advertiser.Close()
+		closeNodes(nodesMap)
+	}()
+
+	p2pBootstrapNodes(nodesMap)
+
+	fmt.Println("Delaying for nodes p2p bootstrap...")
+	time.Sleep(stepDelay)
+
+	round := uint64(0)
+	nonce := uint64(1)
+	round = integrationTests.IncrementAndPrintRound(round)
+	randomness := generateInitialRandomness(1)
+
+	firstNode := nodesMap[0][0]
+
+	transferValue := integrationTests.MinTxGasLimit * integrationTests.MinTxGasPrice / 2
+
+	go func() {
+		for {
+			generateAndSendTxs(
+				firstNode,
+				transferValue,
+				firstNode.OwnAccount.SkTxSign,
+				firstNode.OwnAccount.Address,
+				nodesMap[sharding.MetachainShardId][0].OwnAccount.PkTxSign)
+			time.Sleep(time.Second)
+		}
+	}()
+
+	nbBlocksProduced := uint64(20)
+	var consensusNodes map[uint32][]*integrationTests.TestProcessorNode
+	for i := uint64(0); i < nbBlocksProduced; i++ {
+		printBalance(firstNode)
+
+		for _, nodes := range nodesMap {
+			integrationTests.UpdateRound(nodes, round)
+		}
+		_, _, consensusNodes, randomness = integrationTests.AllShardsProposeBlock(round, nonce, randomness, nodesMap)
+
+		indexesProposers := getBlockProposersIndexes(consensusNodes, nodesMap)
+		integrationTests.SyncAllShardsWithRoundBlock(t, nodesMap, indexesProposers, round)
+		round++
+		nonce++
+
+		time.Sleep(time.Second)
+
+		checkSameBlockHeight(t, nodesMap)
+	}
+
+	printBalance(firstNode)
+}
+
+func setRewardParametersToNodes(nodesMap map[uint32][]*integrationTests.TestProcessorNode) {
+	for _, nodes := range nodesMap {
+		for _, n := range nodes {
+			n.EconomicsData.SetRewards(big.NewInt(0).SetUint64(2 * integrationTests.MinTxGasLimit * integrationTests.MinTxGasPrice))
+		}
+	}
+}
+
+func closeNodes(nodesMap map[uint32][]*integrationTests.TestProcessorNode) {
+	for _, shards := range nodesMap {
+		for _, n := range shards {
+			_ = n.Messenger.Close()
+		}
+	}
+}
+
+func p2pBootstrapNodes(nodesMap map[uint32][]*integrationTests.TestProcessorNode) {
+	for _, shards := range nodesMap {
+		for _, n := range shards {
+			_ = n.Messenger.Bootstrap()
+		}
+	}
+}
+
+func checkSameBlockHeight(t *testing.T, nodesMap map[uint32][]*integrationTests.TestProcessorNode) {
+	for _, nodes := range nodesMap {
+		crtBlock := nodes[0].BlockChain.GetCurrentBlockHeader()
+		for _, n := range nodes {
+			blkc := n.BlockChain.GetCurrentBlockHeader()
+			require.False(t, (crtBlock == nil) != (blkc == nil))
+			require.Equal(t, crtBlock.GetNonce(), blkc.GetNonce())
+		}
+	}
+}
+
+func printBalance(node *integrationTests.TestProcessorNode) {
```

Comment:
you don't print just the balance so please rename this function

Reply:
changed to `printAccount`